model.py

```diff
@@ -313,8 +300,8 @@ def configure_optimizers(self, weight_decay, learning_rate, betas, device_type):
 
         # create the pytorch optimizer object
         optim_groups = [
-            {"params": [param_dict[pn] for pn in sorted(list(decay))], "weight_decay": weight_decay},
-            {"params": [param_dict[pn] for pn in sorted(list(no_decay))], "weight_decay": 0.0},
+            {"params": [param_dict[pn] for pn in sorted(decay)], "weight_decay": weight_decay},
```

Comment:
I have a [related issue](https://github.com/karpathy/nanoGPT/issues/213) that I hope you can explain. Why do we need to sort here?